server/src/main/java/org/opensearch/search/pipeline/PipelinedRequest.java

```diff
@@ -14,21 +14,25 @@
 import org.opensearch.action.search.SearchResponse;
 import org.opensearch.search.SearchPhaseResult;
 
+import java.util.Map;
+
 /**
  * Groups a search pipeline based on a request and the request after being transformed by the pipeline.
  *
  * @opensearch.internal
  */
 public final class PipelinedRequest extends SearchRequest {
     private final Pipeline pipeline;
+    private final Map<String, Object> requestContext;
```

Comment:
instead of having request context as Map<String,Object> should create a class and abstract this map inside the class? Which will make sure that pipelinedRequestContext is acting as container where if needed in future we can add more members without breaking the processor.process interface. 

Also this will ensure that if there are some context variable which we want the individual processors not to change we can have that.

Reply:
> instead of having request context as Map<String,Object> should create a class and abstract this map inside the class? Which will make sure that pipelinedRequestContext is acting as container where if needed in future we can add more members without breaking the processor.process interface later on.

Yeah... leaving it open to potential future changes without breaking existing code sounds like a good idea. 

> Also, it will allow individual processor to extend that PipelineContext class as per their processor needs too.

Not sure what you mean here. We'll need to define `PipelineContext` in core, in order to make sure everyone agrees on the thing we're passing around.

> Also this will ensure that if there are some context variable which we want the individual processors not to change we can have that.

Like values that we want processors to be able to read but not write? (I guess we could also have some "write-once" fields that throw an exception if another processor tries to change it.)

I'll wrap the map in a class (that will just hold the map for now).